Monika After Story/game/script-python.rpy

```diff
@@ -0,0 +1,1205 @@
+
+    if tip_ev.last_seen is None:
+        m 1eub "Now that you know about types, I can teach you about variables."
+
+    # variable intro
+    m 1eua "Variables represent memory locations that store data."
+    m "To create a variable,"
+
+    show monika at t22
+    show screen mas_py_console_teaching
+
+    # a number
+    m 3eua "you do '{b}symbol_name{/b} = {b}value{/b}', like this:"
+
+    call mas_wx_cmd("a_number = " + num_store, local_ctx)
+
+    m "The symbol 'a_number' now points to a memory location storing the integer [num_store]."
+    m "If we enter in the symbol name here,"
+    call mas_w_cmd("a_number")
+    m 3eub "We can retrieve the value that we stored."
+    show monika 3eua
+    call mas_x_cmd(local_ctx)
+
+    m "Notice how we associated the symbol 'a_number' to the value [num_store] using an equals (=) sign?"
+    m 1eub "That is called assignment, where we take whatever is on the left of the equals sign and point it to, or {i}assign{/i} it, the value of whatever is on the right."
+
+    # b_number
+    m 1eua "Assignment is executed in right-to-left order.{w} To illustrate this, let's create a new variable, 'b_number'."
+    call mas_w_cmd("b_number = a_number  -  " + b_num_store)
+
+    m "In assignment, the right side of the equal sign is evaluated first,{w} then its data type is inferred and an appropriate amount of memory is reserved."
+    m "That memory is linked to the symbol on the left via a lookup table."
+    m 1eub "When Python encounters a symbol,{w} it looks that symbol up in the lookup table and replaces it with the value that the symbol was linked to."
+
+    m 3eub "Here, 'a_number' would be replaced with [num_store],{w} so the expression that would be evaluated and assigned to 'b_number' is '[num_store] - [b_num_store]'."
+    show monika 3eua
+    call mas_x_cmd(local_ctx)
+
+    m 1eua "We can verify this by entering only the symbol 'b_number'."
+    m "This will retrieve the value linked to this symbol in the lookup table and show it to us."
+    call mas_wx_cmd("b_number", local_ctx)
+
+    # c number
+    m 3eua "Note that if we enter in a symbol that hasn't been assigned anything, Python will complain with an error."
+    call mas_wx_cmd("c_number", local_ctx)
+
+    m 3eub "But if we assign this symbol a value..."
+    show monika 3eua
+    call mas_wx_cmd("c_number = b_number * a_number", local_ctx)
+    call mas_wx_cmd("c_number", local_ctx)
+
+    m 1hua "Python is able to find the symbol in the lookup table and won't give us an error."
+    
+    m 1eua "The variables we created are all {i}integer{/i} types."
+    m "We didn't have to explicity say that those variables were integers because Python does dynamic typing."
+    m 1eub "This means that the Python interpreter infers the type of a variable based on the data you are storing in it."
+    m "This is unlike other languages like C or Java which require types to be defined with the variable."
```

Comment:
"Other languages, like C or Java, require types to be defined with the variable."